cpp/src/arrow/flight/sql/example/sqlite_server.h

```diff
@@ -0,0 +1,357 @@
+      FlightMessageReader* reader) override;
+
+  Status GetFlightInfoTables(const ServerCallContext& context, const GetTables& command,
+                             const FlightDescriptor& descriptor,
+                             std::unique_ptr<FlightInfo>* info) override;
+
+  Status DoGetTables(const ServerCallContext& context, const GetTables& command,
+                     std::unique_ptr<FlightDataStream>* result) override;
+  Status GetFlightInfoTableTypes(const ServerCallContext& context,
+                                 const FlightDescriptor& descriptor,
+                                 std::unique_ptr<FlightInfo>* info) override;
+  Status DoGetTableTypes(const ServerCallContext& context,
+                         std::unique_ptr<FlightDataStream>* result) override;
+  Status GetFlightInfoImportedKeys(const ServerCallContext& context,
+                                   const GetImportedKeys& command,
+                                   const FlightDescriptor& descriptor,
+                                   std::unique_ptr<FlightInfo>* info) override;
+  Status DoGetImportedKeys(const ServerCallContext& context,
+                           const GetImportedKeys& command,
+                           std::unique_ptr<FlightDataStream>* result) override;
+  Status GetFlightInfoExportedKeys(const ServerCallContext& context,
+                                   const GetExportedKeys& command,
+                                   const FlightDescriptor& descriptor,
+                                   std::unique_ptr<FlightInfo>* info) override;
+  Status DoGetExportedKeys(const ServerCallContext& context,
+                           const GetExportedKeys& command,
+                           std::unique_ptr<FlightDataStream>* result) override;
+  Status GetFlightInfoCrossReference(const ServerCallContext& context,
+                                     const GetCrossReference& command,
+                                     const FlightDescriptor& descriptor,
+                                     std::unique_ptr<FlightInfo>* info) override;
+  Status DoGetCrossReference(const ServerCallContext& context,
+                             const GetCrossReference& command,
+                             std::unique_ptr<FlightDataStream>* result) override;
+
+  Status GetFlightInfoPrimaryKeys(const ServerCallContext& context,
+                                  const GetPrimaryKeys& command,
+                                  const FlightDescriptor& descriptor,
+                                  std::unique_ptr<FlightInfo>* info) override;
+
+  Status DoGetPrimaryKeys(const ServerCallContext& context, const GetPrimaryKeys& command,
+                          std::unique_ptr<FlightDataStream>* result) override;
+
+ private:
+  sqlite3* db_;
+  boost::uuids::random_generator uuid_generator_;
+  std::map<boost::uuids::uuid, std::shared_ptr<SqliteStatement>> prepared_statements_;
```

Comment:
Two things:
1) why use UUIDs? this seems unnecessary, a regular random string should do, and it would avoid an additional Boost dependency
2) better to hide implementation details in the `.cc`, for example using the pimpl idiom:
```c++
struct Impl;
std::shared_ptr<Impl> impl_;
```